backend/dataall/modules/s3_datasets_shares/services/share_managers/share_manager_utils.py

```diff
@@ -19,3 +19,17 @@ def missing_permission_error_msg(requestor, permission_type, permissions, resour
         requestor = ShareErrorFormatter._stringify(requestor)
         permissions = ShareErrorFormatter._stringify(permissions)
         return f'Requestor {requestor} missing {permission_type} permissions: {permissions} for {resource_type} Target: {target_resource}'
+
```

Comment:
Instead of creating a new way of formatting the error message we could:
- for missing permissions use the current `missing_permission_error_msg` and update the permission_type to `'IAM Policy Action'` instead of `'IAM Policy Resource'`
- create a new method for extra permissions `not_allowed_permission_error_msg` that could be re-used in the future (e.g. we could have the case of too many resources granted)